app/src/main/java/com/nextcloud/talk/controllers/ProfileController.java

```diff
@@ -0,0 +1,800 @@
+                R.string.user_info_displayname,
+                Field.DISPLAYNAME,
+                userInfo.getDisplayNameScope());
+        addToList(result,
+                R.drawable.ic_phone,
+                userInfo.getPhone(),
+                R.string.user_info_phone,
+                Field.PHONE,
+                userInfo.getPhoneScope());
+        addToList(result, R.drawable.ic_email, userInfo.getEmail(), R.string.user_info_email, Field.EMAIL, userInfo.getEmailScope());
+        addToList(result,
+                R.drawable.ic_map_marker,
+                userInfo.getAddress(),
+                R.string.user_info_address,
+                Field.ADDRESS,
+                userInfo.getAddressScope());
+        addToList(result,
+                R.drawable.ic_web,
+                DisplayUtils.beautifyURL(userInfo.getWebsite()),
+                R.string.user_info_website,
+                Field.WEBSITE,
+                userInfo.getWebsiteScope());
+        addToList(
+                result,
+                R.drawable.ic_twitter,
+                DisplayUtils.beautifyTwitterHandle(userInfo.getTwitter()),
+                R.string.user_info_twitter,
+                Field.TWITTER,
+                userInfo.getTwitterScope());
+
+        return result;
+    }
+
+    private void addToList(List<UserInfoDetailsItem> info,
+                           @DrawableRes int icon,
+                           String text,
+                           @StringRes int contentDescriptionInt,
+                           Field field,
+                           Scope scope) {
+        info.add(new UserInfoDetailsItem(icon, text, getResources().getString(contentDescriptionInt), field, scope));
+    }
+
+    private void save() {
+        UserInfoAdapter adapter = (UserInfoAdapter) recyclerView.getAdapter();
+
+        for (UserInfoDetailsItem item : adapter.mDisplayList) {
+            // Text
+            if (!item.text.equals(userInfo.getValueByField(item.field))) {
+                String credentials = ApiUtils.getCredentials(currentUser.getUsername(), currentUser.getToken());
+
+                ncApi.setUserData(
+                        credentials,
+                        ApiUtils.getUrlForUserData(currentUser.getBaseUrl(), currentUser.getUserId()),
+                        item.field.fieldName,
+                        item.text)
+                        .retry(3)
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe(new Observer<GenericOverall>() {
+                            @Override
+                            public void onSubscribe(@NotNull Disposable d) {
+                            }
+
+                            @Override
+                            public void onNext(@NotNull GenericOverall userProfileOverall) {
+                                Log.d("ProfileController", "Successfully saved: " + item.text + " as " + item.field);
```

Comment:
use a static TAG for "ProfileController"

Reply:
DONE